### PR TITLE
Support missing tags in XSD [17823]

### PR DESCRIPTION
**Pull request by richiware**
## Description

There are four tags supported by XMLParser but not by XSD file. This PR fixes it. The tags are:

- TimeBasedFilterQos
- PreserntationQos
- DestinationOrderQos
- DurabilityServiceQos

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [ ] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: resources/xsd/fastRTPS_profiles.xsd [hunk 1]
```diff
@@ -1064,7 +1064,7 @@
         ├ disable_heartbeat_piggyback [0~1],
         ├ disablePositiveAcks         [0~1],
         ├ durability                  [0~1],
-        ├ durability_service          [0~1] NOT SUPPORTED YET: COMMENTED,
+        ├ durabilityService           [0~1] NOT SUPPORTED YET: COMMENTED,
```
Comment (JesusPoderoso) [inline resources/xsd/fastRTPS_profiles.xsd:1067]:
```suggestion
        ├ durabilityService           [0~1],
```

File: resources/xsd/fastRTPS_profiles.xsd [hunk 2]
```diff
@@ -1064,7 +1064,7 @@
         ├ disable_heartbeat_piggyback [0~1],
```
Comment (JesusPoderoso) [inline resources/xsd/fastRTPS_profiles.xsd:1064]: Remove `NOT SUPPORTED YET: COMMENTED` comment from previous `destination_order` element.

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Comment by JesusPoderoso** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@mergifyio backport 2.10.x

**Pull request merged**